code/modules/mining/equipment/resonator.dm

```diff
@@ -9,42 +9,60 @@
 	w_class = WEIGHT_CLASS_NORMAL
 	force = 15
 	throwforce = 10
-	var/burst_time = 30
+
+	/// the mode of the resonator; has three modes: auto (1), manual (2), and matrix (3)
+	var/mode = RESONATOR_MODE_AUTO
+	/// How efficient it is in manual mode. Yes, we lower the damage cuz it's gonna be used for mobhunt
+	var/quick_burst_mod = 0.8
+	/// the number of fields the resonator is allowed to have at once
 	var/fieldlimit = 4
+	/// the list of currently active fields from this resonator
 	var/list/fields = list()
-	var/quick_burst_mod = 0.8
+	/// the number that is added to the failure_prob, which is the probability of whether it will spread or not
+	var/adding_failure = 50
+
+/obj/item/resonator/attack_self(mob/user)
+	if(mode == RESONATOR_MODE_AUTO)
+		to_chat(user, "<span class='notice'>You set the [name]'s fields to detonate only after you hit one with it.</span>")
+		mode = RESONATOR_MODE_MANUAL
+	else
+		to_chat(user, "<span class='notice'>You set the [name]'s fields to detonate after 2 seconds.</span>")
+		mode = RESONATOR_MODE_AUTO
 
 /obj/item/resonator/upgraded
 	name = "upgraded resonator"
-	desc = "An upgraded version of the resonator that can produce more fields at once."
+	desc = "An upgraded version of the resonator that can produce more fields at once, as well as having no damage penalty for bursting a resonance field early. It also allows you to set 'Resonance matrixes', that detonate after someone (or something) walks over it."
 	icon_state = "resonator_u"
 	origin_tech = "materials=4;powerstorage=3;engineering=3;magnets=3"
 	fieldlimit = 6
 	quick_burst_mod = 1
 
-/obj/item/resonator/attack_self(mob/user)
-	if(burst_time == 50)
-		burst_time = 30
-		to_chat(user, "<span class='notice'>You set the resonator's fields to detonate after 3 seconds.</span>")
+/obj/item/resonator/upgraded/attack_self(mob/user)
+	if(mode == RESONATOR_MODE_AUTO)
+		to_chat(user, "<span class='notice'>You set [src]'s fields to detonate only after being attacked by [src].</span>")
+		mode = RESONATOR_MODE_MANUAL
+	else if(mode == RESONATOR_MODE_MANUAL)
+		to_chat(user, "<span class='notice'>You set [src]'s fields to work as matrix traps.</span>")
```

Comment:
Maybe name it something like `proximity traps`? `matrix traps` sound very undescriptive

Reply:
![image](https://github.com/user-attachments/assets/5da9ab48-a0a4-4c1a-820e-18f7f4104538)
this already mention on upgraded resonator description, not sure if it need to change into something else.